chapters/interface.tex

```diff
@@ -537,6 +537,11 @@ \section{Function-Compatibility or Function-Subtyping for Functions}\label{funct
   constraining interface of the function being redeclared.
 \end{itemize}
 
+Note that function-compatibility does currently not ensure that the function call has the correct variability, and that must then be checked after re-declaration -- due to the possibility of default arguments with higher variability.
```

Comment:
This is an important note, but I think we should aim for a situation where we don't need to say that our variability definitions are wrong.  How about something like this (combined with additional changes in _Variability of Expressions_ to be suggested next):
```suggestion
Note that variability of function calls cannot be determined using the just the interface of a function, as the variabilities of default argument expressions are not expressed by the interface.
Hence a function redeclaration being function-compatible does not ensure that function calls will fulfill variability requirements, and tools must therefore check variability requirements separately.

\begin{nonnormative}
One way that tools may carry out the additional checks is to first infer the variability requirements on functions that can be redeclared, but looking at the expressions where the function is called.
The interface of the function that could be redeclared can then be extended with a variability requirement for any default argument expressions, allowing errors to be detected and reported at the point of a redeclaration that introduces too high variability through default argument expressions.
\end{nonnormative}
```

By the way, we only have _re-declaration_ written with a hyphen in one other place, and we should probably get rid of the hyphen also in that case.

Reply:
there is a the too much here

Reply:
Most of this was merged, but the non-normative part reduced; so I think this is resolved.

Reply:
> there is a the too much here

Fixed.

Reply:
> Most of this was merged, but the non-normative part reduced; so I think this is resolved.

To me, the non-normative part looks stripped rather than reduced.  I'd like to leave this conversation open, as the non-normative part is closely related to the ongoing discussion below in https://github.com/modelica/ModelicaSpecification/pull/2924#discussion_r642131665.

Reply:
I think we can handle that later, and merge this first as it is difficult to keep track of everything.